python installation requires root privilege

**Avogadro version: (please complete the following information from the About box):**
 - Avogadrolibs: 9e08b9b51c37091775df71d30b08fc68c4a77b46
 - Qt: 6.10.0

**Desktop version: (please complete the following information):**
 - OS: EndeavourOS
 - Compiler: GCC 15.2.1
 - Python 3.13.7

**Describe the bug**

https://github.com/OpenChemistry/avogadrolibs/blob/9e08b9b51c37091775df71d30b08fc68c4a77b46/python/CMakeLists.txt#L35
https://github.com/OpenChemistry/avogadrolibs/blob/9e08b9b51c37091775df71d30b08fc68c4a77b46/python/CMakeLists.txt#L48
https://github.com/OpenChemistry/avogadrolibs/blob/9e08b9b51c37091775df71d30b08fc68c4a77b46/python/CMakeLists.txt#L64

avogadro installs python components under `${Python_SITEARCH}`

https://github.com/OpenChemistry/avogadrolibs/blob/9e08b9b51c37091775df71d30b08fc68c4a77b46/python/CMakeLists.txt#L20

i.e. `/usr/lib/python3.13/site-packages`, which is a system directory and unrelated to `CMAKE_INSTALL_PREFIX`.
This requires root privilege.

```
$ cmake --install .
…
CMake Error at python/cmake_install.cmake:60 (file):
  file cannot create directory: /usr/lib/python3.13/site-packages/avogadro.
  Maybe need administrative privileges.
Call Stack (most recent call first):
  cmake_install.cmake:67 (include)
```

**To Reproduce**

1. Build avogadro with `-DUSE_PYTHON=ON`
2. `cmake --install .` without root privilege

**Expected behavior**

Python files are installation under `${CMAKE_INSTALL_PREFIX}/lib/python3.13/site-packages/avogadro`, while a user has to set PYTHONPATH.

--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -17,7 +17,16 @@ set(wrapper_SRCS
 set(_python_module_install_dir "avogadro")
 # SKBUILD is set for binary wheel
 if (NOT SKBUILD)
-  set(_python_module_install_dir "${Python_SITEARCH}/avogadro")
+  execute_process(COMMAND "${Python_EXECUTABLE}" -c [=[
+import os
+import sysconfig
+print(sysconfig.get_path(name='platlib', vars={'platbase': ''}).lstrip(os.sep), end='')
+# this prints e.g. 'lib/pythonX.Y/site-packages' on a Linux machine,
+# which is relative so that CMake `file(INSTALL)` prepends ${CMAKE_INSTALL_PREFIX}
+]=]
+    OUTPUT_VARIABLE PYTHON_LIBDIR
+  )
+  set(_python_module_install_dir "${PYTHON_LIBDIR}/avogadro")
 endif()
 
 set(CMAKE_MODULE_LINKER_FLAGS "")

